src/actions/guides/database/models.cr

```diff
@@ -7,104 +7,37 @@ class Guides::Database::Models < GuideAction
-    # src/models/user.cr
-    class User < BaseModel
-      table :users do
-        column name : String
-        column age : Int32?
-      end
-    end
-    ```
+    ### Defining a column
 
-    > You can add a `?` to the end of any type to mark it as nilable.
+    show all options, and example of each supported type
 
-    ### Column types
+    Default generated columns
 
-    You can use the following types in your columns.
+    Nillable vs non-nillable types
 
-    * `String` - `text` column type. In Postgres [`text` can store strings of any length](https://stackoverflow.com/questions/4848964/postgresql-difference-between-text-and-varchar-character-varying)
-    * `Int32` - `int` column type.
-    * `Time` - `timestamp` column type.
-    * `Float64` - `decimal` column type.
-    * `Bool` - `bool` column type.
+    ## Model Associations
 
-    ## Using associations
+    What they are...
 
-    Right now you can define `has_many`, `has_one`, and `belongs_to` associations.
+    > Note that adding require statements in to the model may be necessary due to how crystal require works?
 
-    ```crystal
-    # Require the models we want to associate. If we don't we may get an
-    # "Undefined constant" error because Lucky didn't load the associated models yet
-    require "./supervisor"
-    require "./task"
-    require "./company"
+    ### Belongs to
 
-    class User < BaseModel
-      table :users do
-        has_one supervisor : Supervisor
-        has_many tasks : Task
-        belongs_to company : Company
-      end
-    end
+    ### Has One (one to one)
 
-    # Will return the company associated with the User
-    UserQuery.new.preload_company.find(1).company
-    ```
+    ### Has Many (one to many)
 
-    ### Making associations optional
+    ### Has Many Through (many to many)
 
-    Sometimes associations are not required. To do that add a `?` to the end of the type.
-
-    ```crystal
-    belongs_to company : Company?
-    ```
-
-    > Make sure to make the column nilable in your migration as well: `belongs_to Company?`
-
-    ### Has many through (many-to-many)
-
-    Let's say we want to have many tags that can belong to any number of posts.
-
-    Here are the models:
-
-    ```crystal
-    # This is what will join the posts and tags together
-    class Tagging < BaseModel
-      table :taggings do
-        belongs_to tag : Tag
-        belongs_to post : Post
-      end
-    end
-
-    class Tag < BaseModel
-      table :tags do
-        column name : String
-        has_many taggings : Tagging
-        has_many posts : Post, through: :taggings
-      end
-    end
-
-    class Post < BaseModel
-      table :posts do
-        column title : String
-        has_many taggings : Tagging
-        has_many tags : Tag, through: :taggings
-      end
-    end
-    ```
-
-    > The associations *must* be declared on both ends (the Post and the Tag in
-    this example), otherwise you will get a compile time error
+    ### Polymorphic?
```

Comment:
Not yet officially supported in an easy to use way but can be done in a more custom way. We can maybe add something later as this might take a bit more trial and error to figure out a good suggestion